rts/motoko-rts/src/leb128.rs

```diff
@@ -65,23 +65,23 @@ pub unsafe fn leb128_decode_checked(buf: *mut Buf) -> Option<u32> {
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn sleb128_decode(buf: *mut Buf) -> i32 {
+pub unsafe extern "C" fn sleb128_decode(buf: *mut Buf) -> isize {
     sleb128_decode_checked(buf).expect("sleb128_decode: overflow")
 }
 
 /// Returns `None` on overflow
-pub unsafe fn sleb128_decode_checked(buf: *mut Buf) -> Option<i32> {
+pub unsafe fn sleb128_decode_checked(buf: *mut Buf) -> Option<isize> {
     let mut result = 0;
     let mut shift = 0;
 
     let last_byte = loop {
         let byte = read_byte(buf);
 
-        result |= ((byte & 0b0111_1111) as i32) << shift;
+        result |= ((byte & 0b0111_1111) as isize) << shift;
 
         // Overflow check ported from Wasm reference implementation:
         // https://github.com/WebAssembly/spec/blob/f9770eb75117cac0c878feaa5eaf4a4d9dda61f5/interpreter/binary/decode.ml#L89-L98
-        if shift == 28 && (byte & 0b0111_1000 != 0 && byte & 0b0111_1000 != 0b0111_1000) {
+        if shift == 63 && (byte & 0b0111_1111 != 0 && byte & 0b0111_1111 != 0b0111_1111) {
```

Comment:
I'll just trust you on this one....

Reply:
perhaps @ggreif could review this bit?

Reply:
Yes, a review by @ggreif would be great. This is a quite complicated logic.